File: woodroid/src/main/java/me/gilo/woodroid/repo/product/ReviewRepository.java
package me.gilo.woodroid.repo.product;

import me.gilo.woodroid.data.api.ProductReviewAPI;
import me.gilo.woodroid.models.ProductReview;
import me.gilo.woodroid.models.filters.ProductReviewFilter;
import me.gilo.woodroid.repo.WooRepository;
import retrofit2.Call;

import java.util.List;

public class ReviewRepository extends WooRepository {

    private final ProductReviewAPI apiService;

    public ReviewRepository(String baseUrl, String consumerKey, String consumerSecret) {
       super(baseUrl, consumerKey, consumerSecret);
       apiService = retrofit.create(ProductReviewAPI.class);
    }

    public Call<ProductReview> create(ProductReview review) {
        return apiService.create(review);
    }


    public Call<ProductReview> review(int id) {
        return apiService.view(id);
    }

    public Call<List<ProductReview>> reviews() {
        return apiService.list();
    }

    public Call<List<ProductReview>> reviews(ProductReviewFilter productReviewFilter) {
        return apiService.filter(productReviewFilter.getFilters());
    }

    public Call<ProductReview> update(int id, ProductReview review) {
        return apiService.update(id, review);
    }

    public Call<ProductReview> delete(int id) {
        return apiService.delete(id);
    }

    public Call<ProductReview> delete(int id, boolean force) {
        return apiService.delete(id, force);
    }


}

File: woodroid/src/main/java/me/gilo/woodroid/repo/product/ReviewRepository.kt
package me.gilo.woodroid.repo.product

import me.gilo.woodroid.data.api.ProductReviewAPI
import me.gilo.woodroid.models.ProductReview
import me.gilo.woodroid.models.filters.ProductReviewFilter
import me.gilo.woodroid.repo.WooRepository
import retrofit2.Call

class ReviewRepository(baseUrl: String, consumerKey: String, consumerSecret: String) :
    WooRepository(baseUrl, consumerKey, consumerSecret) {

    private val apiService: ProductReviewAPI

    init {
        apiService = retrofit.create(ProductReviewAPI::class.java)
    }

    fun create(review: ProductReview): Call<ProductReview> {
        return apiService.create(review)
    }


    fun review(id: Int): Call<ProductReview> {
        return apiService.view(id)
    }

    fun reviews(): Call<List<ProductReview>> {
        return apiService.list()
    }

    fun reviews(productReviewFilter: ProductReviewFilter): Call<List<ProductReview>> {
        return apiService.filter(productReviewFilter.filters)
    }

    fun update(id: Int, review: ProductReview): Call<ProductReview> {
        return apiService.update(id, review)
    }

    fun delete(id: Int): Call<ProductReview> {
        return apiService.delete(id)
    }

    fun delete(id: Int, force: Boolean): Call<ProductReview> {
        return apiService.delete(id, force)
    }


}
